### PR TITLE
fix/8389

### LINKED ISSUE #8389: Guards dont follow player
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Version?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.
I am also running the latest (possible) versions of other mods that are part of my problem.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies/Structurize [Wiki](https://wiki.minecolonies.ldtteam.com/) or I asked on [discord](https://discord.minecolonies.com/). Or I was sent from discord to open an issue here.

### Minecraft Version

1.18

### MineColonies Version

1.0.747-ALPHA

### Structurize Version

1.0.399-RELEASE

### Related Mods and their Versions

1. blockui-1.18.2-0.0.47-ALPHA
2. domum_ornamentum-1.18.2-1.0.50-ALPHA-universal
3. minecolonies-1.18.2-1.0.747-ALPHA
4. multi-piston-1.18.2-1.2.7-ALPHA
5. structurize-1.18.2-1.0.399-RELEASE

### Current Behavior

Guards dont follow player in "follow mode", they just stand still and teleport to player position, if player walk too far from guard.

### Expected Behavior

Guards must follow player in follow mode.

### Reproduction Steps

1. build townhall
2. create colony
3. build builder structure
4. build guard tower
5. set the follow mode in guard tower menu

### Logs

https://gist.github.com/KianAshum/21e26f121d874b65d4886a8a69ee56d9#file-gistfile1-txt

### Anything else?

_No response_

### Footer

---
## Viewers
* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by ImEr1c**
Closes #8389 

# Changes proposed in this pull request:
- NBT serialize and deserialize for followPlayerUUID

Review please

### CODE REVIEW

**Comment by Raycoms** [comment]
So, your issue is that after a relog the guard stopped following you?

**Comment by ImEr1c** [comment]
yeah, i checked multiple times.. that was the issue

**Comment by ImEr1c** [comment]
so is this ok? or should I make changes?

**Comment by Thodor12** [comment]
> so is this ok? or should I make changes?

Just wait until one of the devs reviews your code

**Comment by ImEr1c** [comment]
ok

**Comment by ImEr1c** [comment]
review..?

**Comment by MotionlessTrain** [comment]
> Just wait until one of the devs reviews your code

^ That part hasn't changed. The devs are probably busy, and haven't had time yet to look at your code in depth

**Comment by ImEr1c** [comment]
ok..

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java [hunk 1]
```diff
@@ -37,6 +37,7 @@
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.Tuple;
+import net.minecraft.util.UUIDCodec;
```
Comment (Thodor12) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java]: Just remove the import as it's no longer used

**Review by Raycoms — APPROVED** [review]

**Review by someaddons — APPROVED** [review]

**Pull request merged**